hnn_core/batchsimulate.py

```diff
@@ -0,0 +1,148 @@
+"""Batch simulation."""
+from joblib import Parallel, delayed
+from hnn_core import simulate_dipole
+from hnn_core.network_models import (jones_2009_model,
+                                     calcium_model, law_2021_model)
+
+
+class BatchSimulate:
+    def __init__(self, set_params, net_name='jones', tstop=170,
+                 dt=0.025, n_trials=1):
+        """
+        Initialize the BatchSimulate class.
+
+        Parameters
+        ----------
+        set_params : func
+            User-defined function that sets parameters in network drives.
+            `set_params(net, params) -> None`
+        net_name : str, optional
+            The name of the network model to use. Default is 'jones'.
+        tstop : float, optional
+            The stop time for the simulation. Default is 170 ms.
+        dt : float, optional
+            The time step for the simulation. Default is 0.025 ms.
+        n_trials : int, optional
+            The number of trials for the simulation. Default is 1.
+        """
+        self.set_params = set_params
+        self.net_name = net_name
+        self.tstop = tstop
+        self.dt = dt
+        self.n_trials = n_trials
+
+    def run(self, param_grid, max_size=None, return_output=True,
+            save_output=False, fpath='./', n_jobs=-1):
```

Comment:
n_jobs = -1 is dangerous ... you'll choke your computer. Default should always be 1